### PR TITLE
rails 3.2

### PR DESCRIPTION
**Pull request by abenari**

### CODE REVIEW

**Comment by isratrade** [comment]
I'm about done with the integration tests.  I will merge it and run it against your branch.

**Comment by isratrade** [comment]
@abenari I made 5 small commits at my branch https://github.com/isratrade/foreman/tree/rails32_amos that you can merge or cherry-pick into your branch.

I ran the integration tests and ..... results please .....
1)  All non-javascript tests passed - yea!!! 
2)  JS failures were:
             \* create new Compute Resource - fields don't appear after selecting provider
             \* create new User - fields don't appear after selecting Auth Source

Even though the integration tests take time, it's much faster than a person clicking through the site.

File: Gemfile [hunk 1]
```diff
@@ -2,14 +2,13 @@ require File.expand_path('../config/settings', __FILE__)
...
-gem 'rails', '3.0.20'
-gem "jquery-rails"
+gem 'rails', '3.2.13'
 gem 'json'
 gem 'rest-client', :require => 'rest_client'
 gem "audited-activerecord", "3.0.0.rc1"
 gem "will_paginate", "~> 3.0.2"
 gem "ancestry", "~> 1.3"
-gem 'scoped_search', '>= 2.4'
+gem 'scoped_search', :git => "git://github.com/wvanbergen/scoped_search.git"
```
Comment (xsuchy) [inline Gemfile:11]: Do we really need last version from git? Isn't version 2.4.1 from March 6 enough?

**Comment by lzap** [comment]
One comment - the Gemfile change now requires scoped_search to be repacked from git. For Katello this is an extra effort, is it feasible to release scoped_search version until this is merged? Katello needs always an RPM installation for Foreman, they just do not run it from git.

**Comment by isratrade** [comment]
@amos maintains the scoped_search gem and simply hasn't pushed the new version to rubygems.  The :git => path is just temporary.

**Comment by lzap** [comment]
I understand all of that :-) My appeal is still valid tho. Thanks! ;-)

**Comment by lzap** [comment]
Maybe to explain my motivation - Katello team is now waiting for this
branch to be merged so they can move on to Rails 3.2. Also they have a
bunch of work on SCL (Software Collection stack) for Ruby 1.9 Rails 3.2
for RHEL6 compatible distros. That makes things more complicated there.

If it is possible just to bump version of scoped search, could save some
more packaging work.

Thanks

**Comment by GregSutcliffe** [comment]
+1 for avoiding git gems wherever possible - it affects the Debian packaging too. If it's essential, so be it, but for situations like this (where we control the upstream gems) can we avoid putting this stuff into develop?

**Comment by xsuchy** [comment]
```
ActionView::Template::Error (undefined method `update_page' for #<#<Class:0x00000006771ab8>:0x00000006759760>):
    8: <p>
    9:   <%= _("If you feel this is an error with Foreman itself, please open a new issue with") %> <%= link_to _("Foreman ticketing system"), "http://theforeman.org/projects/foreman/issues", :rel => "external" %>,
    10:   <%= _("You would probably need to attach the") %>
    11:   <%= link_to_function _("Full trace"), toggle_div("backtrace") %> <%= _("and relevant log entries.") %>
    12: </p>
    13: 
    14: <div id="backtrace" class="alert alert-message alert-block alert-error base in fade">
  app/helpers/application_helper.rb:42:in `toggle_div'
  app/views/common/500.html.erb:11:in `_app_views_common_____html_erb__3714896745807706039_52547440'
  app/controllers/application_controller.rb:291:in `generic_exception'
```

This does not work under Rails 3.2
http://apidock.com/rails/ActionView/Helpers/PrototypeHelper/update_page

**Comment by ares** [comment]
I have a fix for error discovered by @xsuchy - I rewrote a code generating Prototype JS to generate jQuery instead. Check my fork of this PR https://github.com/ares/foreman/tree/pr/467

**Comment by ohadlevy** [comment]
Thanks Marek, but its already fixed in isratrade branch, we are currently
testing them..

**Comment by ares** [comment]
I think he forgot to fix this one (https://github.com/isratrade/foreman/blob/rails32_amos/app/helpers/application_helper.rb#L41) so my patch should be applied after his patches.

**Comment by ohadlevy** [comment]
he simply didnt push it yet - sorry :)

**Comment by ares** [comment]
oh I see, np :-)

**Comment by xsuchy** [comment]
I could not test isratrade patch, because it is not public, but with @ares patch I get further and now I'm getting:
```ActionView::Template::Error (application.css isn't precompiled):
    2: <html>
    3:   <head>
    4:     <title> <%= h(yield(:title) || "Foreman") %></title>
    5:     <%= stylesheet_link_tag 'application' %>
    6:     <%= javascript_include_tag 'application' %>
    7:     <%= csrf_meta_tags %>
    8:     <%= javascript_tag "var AUTH_TOKEN = #{form_authenticity_token.inspect};" if protect_against_forgery? %>
  app/views/layouts/application.html.erb:5:in`_app_views_layouts_application_html_erb___1785201843474707935_41610880'
  app/controllers/application_controller.rb:291:in `generic_exception'

```
```

**Comment by isratrade** [comment]
new PR https://github.com/theforeman/foreman/pull/487 and closing this one.
@ohadlevy @xsuchy @ares @GregSutcliffe @abenari

**Comment by abenari** [comment]
new PR #487 replace this one.

**Pull request closed without merging**